### PR TITLE
Added Pagination

### PR DESCRIPTION
**Pull request by Shrutichhatbar**

### CODE REVIEW

File: Craigslist/Craigslist.Business/Content/PagedList.css [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+.pager .previous > a,
+.pager .previous > span {
+  float: left;
+}
+
+.pager .disabled > a,
+.pager .disabled > a:hover,
+.pager .disabled > a:focus,
+.pager .disabled > span {
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Business/Content/PagedList.css]: @Shrutichhatbar  Where all these styles come from ?
From PagesList.mvc package?
Do you use them?

File: Craigslist/Craigslist.Domain/Content/PagedList.css [hunk 2]
```diff
@@ -0,0 +1,166 @@
+.pagination {
+  display: inline-block;
+  padding-left: 0;
+  margin: 20px 0;
+  border-radius: 4px;
+}
+
+.pagination > li {
+  display: inline;
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Domain/Content/PagedList.css]: Why do you have css in domain!!!!!?????? O_o
And it's probably the same authogenerated css from paged list package.

File: Craigslist/Craigslist.Domain/Craigslist.Domain.csproj [hunk 3]
```diff
@@ -36,6 +36,12 @@
     <Reference Include="EntityFramework.SqlServer">
       <HintPath>..\packages\EntityFramework.6.1.3\lib\net45\EntityFramework.SqlServer.dll</HintPath>
     </Reference>
+    <Reference Include="PagedList">
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Domain/Craigslist.Domain.csproj]: Why does Domain need this package ?

File: Craigslist/Craigslist.Selenium/Content/PagedList.css [hunk 4]
```diff
@@ -0,0 +1,166 @@
+.pagination {
+  display: inline-block;
+  padding-left: 0;
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Selenium/Content/PagedList.css]: The same css in Selenium project. It definitely has nothing to do with UI test automation!

File: Craigslist/Craigslist.Selenium/Craigslist.Selenium.csproj [hunk 5]
```diff
@@ -32,6 +32,12 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="PagedList">
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Selenium/Craigslist.Selenium.csproj]: You don't need this package in Selenium. why did you add it here?
Did you see the changes before pull request?

File: Craigslist/Craigslist.Selenium/packages.config [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Selenium/packages.config]: The same

File: Craigslist/Craigslist.Tests/Content/PagedList.css [hunk 7]
```diff
@@ -0,0 +1,166 @@
+.pagination {
+  display: inline-block;
+  padding-left: 0;
+  margin: 20px 0;
+  border-radius: 4px;
+}
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Tests/Content/PagedList.css]: The Same
remove

File: Craigslist/Craigslist.Tests/Craigslist.Tests.csproj [hunk 8]
```diff
@@ -34,6 +34,12 @@
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework" />
+    <Reference Include="PagedList">
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Tests/Craigslist.Tests.csproj]: The same

File: Craigslist/Craigslist.Tests/app.config [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Tests/app.config:3]: Did you see that?? O_o

File: Craigslist/Craigslist/Content/PagedList.css [hunk 10]
```diff
@@ -0,0 +1,166 @@
+.pagination {
+  display: inline-block;
+  padding-left: 0;
+  margin: 20px 0;
+  border-radius: 4px;
+}
+
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist/Content/PagedList.css]: The same

File: Craigslist/Craigslist/Controllers/ListingController.cs [hunk 11]
```diff
@@ -13,9 +14,11 @@ public class ListingController : Controller
 		private readonly LookupManager lookupManager = new LookupManager();
 		private readonly CategoriesHelper categoriesHelper = new CategoriesHelper();
 
-	    public ViewResult List(long? categoryId = null)
+        public ViewResult List(int? page, long? categoryId = null )
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist/Controllers/ListingController.cs:16]: Alignment issue

File: Craigslist/Craigslist.Domain/Content/PagedList.css (hunk 2 above)
Comment (Shrutichhatbar) [inline Craigslist/Craigslist.Domain/Content/PagedList.css]: everything came from pageList.mvc nuget package

File: Craigslist/Craigslist.Business/Content/PagedList.css (hunk 1 above)
Comment (Shrutichhatbar) [inline Craigslist/Craigslist.Business/Content/PagedList.css]: Actually it works without that.. but it was installed automatically when I installed pagelist.mvc

File: Craigslist/Craigslist/Controllers/ListingController.cs [hunk 12]
```diff
@@ -13,9 +14,11 @@ public class ListingController : Controller
...
 	    {
-		    return View(listingsManager.GetListingsByCategoryId(categoryId));
+            int pageSize = 1;
+            int pageNumber = (page ?? 1);
+		    return View(listingsManager.GetListingsByCategoryId(categoryId).ToPagedList(pageNumber, pageSize));
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist/Controllers/ListingController.cs]: Do you understand what you are doing here?
Imaging you have 100000 items. You're fetching all of them into server memory and then applying pagination. 
i'll take a look at the implementation of ToPagedList method but i'm pretty sure that it just applies Skip and Take LINQ methods behind the scene. 
It's not acceptable. 
Skip and Take should be applied withing the opened domain context in order to get translated into SQL top() and row_number() instructions.

Comment (AlexM-Itu) [inline Craigslist/Craigslist/Controllers/ListingController.cs]: btw, why is this logic in the controller? why not in the Manager

File: Craigslist/Craigslist.Business/ListingsManager.cs [hunk 13]
```diff
@@ -17,17 +21,20 @@ public void PublishListing(Listing listing)
 			}
 		}
 
-		public List<Listing> GetListingsByCategoryId(long? categoryId)
+		public IPagedList<Craigslist.Domain.Entities.Listing> GetListingsByCategoryId(int? page, long? categoryId)
 		{
 			using (var domain = new CraigslistDomain())
 			{
+                int pageSize = Convert.ToInt32(ConfigurationManager.AppSettings["pageSize"]);
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Business/ListingsManager.cs]: in's totally OK to use methods from class Convert, but for parsing most of standard types there're shortcuts like int.Parse(...) or long.Parse(...) or DateTime.Parse(...).. It's just shorter :)

File: Craigslist/Craigslist.Business/ListingsManager.cs [hunk 14]
```diff
@@ -17,17 +21,20 @@ public void PublishListing(Listing listing)
...
+                int pageNumber = (page ?? 1);
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Business/ListingsManager.cs:29]: Is the index 1 or 0 based?

File: Craigslist/Craigslist.Business/ListingsManager.cs [hunk 15]
```diff
@@ -17,17 +21,20 @@ public void PublishListing(Listing listing)
...
 				return domain
 					.Listings
 					.Include(l => l.Category)
 					.Include("Category.ParentCategory")
 					.Include(l => l.Contact)
-					.Where(listing => categoryId == null || listing.CategoryId == categoryId)
-					.ToList();
+                    .Where(listing => categoryId == null || listing.CategoryId == categoryId)
+                    .OrderBy(x => x.Created)
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Business/ListingsManager.cs]: for lambdas there're some naming conventions. Regularly they're called i, j, k or by first letter of the entity (l in my case means Listing) or by its full name (Listing).

File: Craigslist/Craigslist.Business/ListingsManager.cs [hunk 16]
```diff
@@ -17,17 +21,20 @@ public void PublishListing(Listing listing)
...
+					.ToPagedList(pageNumber, pageSize);
```
Comment (AlexM-Itu) [inline Craigslist/Craigslist.Business/ListingsManager.cs:37]: :+1:
